### ISSUE TITLE
VS Jupiter kernel crashes in debug mode

### ISSUE THREAD

**Issue by deep-learning-newbie** [microsoft/vscode#193409]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 
- OS Version: 

Steps to Reproduce:

1. Connect VSCode to the remote system using SSH connection
2. enter debug session
3. point mouse at the variable
4. debug session will be crashed and you can see message  - Server[1] disconnected unexpectedly 
5. the same in jupyter cell debug 

My operation system windws 11. remote OS Ubuntu 20.
VSCode version 
Version: 1.82.2 (user setup)
Commit: abd2f3db4bdb28f9e95536dfa84d8479f1eb312d
Date: 2023-09-14T05:55:25.390Z
Electron: 25.8.1
ElectronBuildId: 23779380
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.22621

extensions: 
- Jupyter  v2023.8.1002501831 
- Python  v2023.16.0

**Comment by roblourens** [comment]
I would guess that this is a jupyter/debugpy issue and you would see the same when debugging with jupyterlab? Can you check?

**Comment by deep-learning-newbie** [comment]
sorry i don't understand what you mean - how i can check it in Jupyter lab- do you mean %whos or locals() %debug or do you mean something else.  i can't point mouse at variable and watch  its value in Jupyter lab. also I can't expand variables in jyputer - that's why i use vscode. could you provide me instruction what should i do, or install, or setup to be able to provide something useful for you ?

**Comment by deep-learning-newbie** [comment]
- whan I point over mouse at variable 
<img width="435" alt="image" src="https://github.com/microsoft/vscode/assets/63535316/15d41080-6ebd-459a-b379-c933b064383b">

- when i expand the variable in watch window will crash
<img width="292" alt="image" src="https://github.com/microsoft/vscode/assets/63535316/e274ec96-f0fa-4b0e-9f2c-3fb47074050e">

**Comment by roblourens** [comment]
Would you know anything about this @DonJayamanne?

**Comment by deep-learning-newbie** [comment]
No, but if you want I install something to test it to figure out the reason of this issue - just let me know. I'm opened because it really hard to develop without debugging and this issue make me mad.

**Comment by DonJayamanne** [comment]
@deep-learning-newbie 
I too can replicate this at my end. Will look into this. 

The only logs we have are 
```
14:13:56.008 [debug] [Debug] to kernel: {"command":"variables","arguments":{"variablesReference":15},"type":"request","seq":41}
14:13:56.008 [debug] [Debug] to kernel, mapped: {"command":"variables","arguments":{"variablesReference":15},"type":"request","seq":41}
14:13:56.098 [debug] KernelProcess Exited, Exit Code - undefined 
14:13:56.098 [error] Disposing session as kernel process died ExitCode: undefined, Reason: 
14:13:56.099 [info] Dispose Kernel process 33599.
14:13:56.102 [debug] Disposed Kernel process 33599.
14:13:56.116 [info] End cell 1 execution @ undefined, started @ 1695269626095, elapsed time = -1695269626.095s
14:13:56.121 [debug] Executing silently Code (dead) = import os as _VSCODE_os\n_VSCODE_fileList = ["/var/folders/fv/524cyb9j03s16612v0f52g4h0000gn/T/ipyker
14:13:56.133 [info] Skipping sending message request because session is disposed

```

**Comment by DonJayamanne** [comment]
Crashes in regular Python as well, i.e. this isn't an issue with Jupyter ntoebooks,
